src/Riok.Mapperly/Descriptors/InlineExpressionRewriter.cs

```diff
@@ -55,6 +55,17 @@ public override SyntaxNode VisitTypeArgumentList(TypeArgumentListSyntax node)
         return node.WithArguments(SeparatedList(args));
     }
 
+    public override SyntaxNode? VisitMemberAccessExpression(MemberAccessExpressionSyntax node)
+    {
+        if (semanticModel.GetSymbolInfo(node.Expression).Symbol is ITypeSymbol namedTypeSymbol)
+        {
+            var expression = FullyQualifiedIdentifier(namedTypeSymbol).WithTriviaFrom(node.Expression);
+            node = node.WithExpression(expression);
```

Comment:
Wouldn't it safer to return the new node here directly instead of visiting it on L66? Otherwise this could result in an exception that the new node is not part of the syntax tree, couldn't it?

Reply:
I do not have an experience with visitor pattern implementation. I changed it to return directly